TripalCultivate/TripalCultivate-Phenotypes
# G0.06 configuration ontology and terms

**Issue #[6] G0.06 Configuration Form for Ontology and Terms**
https://github.com/TripalCultivate/TripalCultivate-Phenotypes/issues/6
## Motivation


Create genus ontology and terms configuration page that will define vocabulary terms and controlled vocabulary terms used to house traits and by many processes in the phenotypes module.

## What does this PR do?
*Please describe each things this PR does. For example, a PR may 1) solve a specific bug, 2) create an auomated test to ensure it doesn't return.*

1. Create configuration page
2. Functional test that tests page access permission, existence of fields in the page/form, set value to field and submit the form and posts warning message when Tripal Job to install terms and genus ontology was not executed.

## Testing

### Automated Testing
*Please describe each automated test this PR creates and provide a list of the assertions it makes using casual language.*
*Do not just say things like "asserts the array is not empty" but rather say "Ensures that the return value of method X with these parameters is not an empty array".*

1. Asserts page is accessible with administrative/tripal access permission.
2. Asserts page loads with response status code of 200.
3. Asserts page posts warning when terms and genus ontology are not installed through Tripal job setup during install.
4. Asserts page load when terms and genus ontology are installed (manually created).
5. Asserts for all sample genus and terms have correct field.
6. Asserts setting a value to all fields for all sample genus and terms and submitting the form, that values set are saved.
7. Asserts that retrieving configuration values for all sample genus and terms match the value set.
8. Asserts allow new traits field saves true or false values.

ClassName->testSpecificBug: tests this specific bug
- 
- 
- 

### Manual Testing
*Describe in detail how someone should manually test this functionality.*
*Make sure to include whether they need to build a docker from scratch, create any records, etc.*

1. Install Tripal Cultivate Phenotypes - trpcultivate_phenotypes module.
2. If your build has no genus and accessing the configuration page - my site/admin/tripal/extension/tripal-cultivate/phenotypes/ontology will instruct to execute tripal job. A link to to tripal job page and the job title are encoded into the warning message.
3. Create sample Genus and execute the Tripal Job mentioned.
4. Clear cache and refresh Genus Ontology and Terms configuration page.
